### PR TITLE
Identity validation & minor improvements

### PR DESCRIPTION
**Pull request by mdomasevicius**

### CODE REVIEW

File: tequilapi/contract/identity.go [hunk 1]
```diff
@@ -129,6 +130,25 @@ type IdentityRegisterRequest struct {
 	ReferralToken *string `json:"token,omitempty"`
 }
 
+// Validate validates fields in request
+func (irr *IdentityRegisterRequest) Validate() *validation.FieldErrorMap {
+	errors := validation.NewErrorMap()
+
+	if okAddress := common.IsHexAddress(irr.Beneficiary); irr.Beneficiary != "" && !okAddress {
```
Comment (Waldz) [inline tequilapi/contract/identity.go:137]: Shouldn't `irr.Beneficiary != ""` got as very first clause?

File: tequilapi/contract/identity.go [hunk 2]
```diff
@@ -129,6 +130,25 @@ type IdentityRegisterRequest struct {
...
+		errors.ForField("beneficiary").Invalid(irr.Beneficiary+" - is not a valid ethereum wallet address")
+	}
+
+	if irr.ReferralToken == nil {
+		if irr.Stake == nil {
+			errors.ForField("stake").Required()
+		} else if irr.Stake.Cmp(big.NewInt(0)) == -1 {
```
Comment (Waldz) [inline tequilapi/contract/identity.go:144]: @chompomonim, @vkuznecovas what is minimal stake for provider to register?

File: tequilapi/contract/pageable.go [hunk 3]
```diff
@@ -55,20 +54,9 @@ func (q *PaginationQuery) Bind(request *http.Request) *validation.FieldErrorMap
...
-	}
-	if qStr := qs.Get("page"); qStr != "" {
-		if qVal, err := parseInt(qStr); err != nil {
-			errs.ForField("page").Add(err)
-		} else {
-			q.Page = *qVal
-		}
-	}
+
+	NewQueryParse(qs, "page_size").BindInt(&q.PageSize, errs)
```
Comment (Waldz) [inline tequilapi/contract/pageable.go:58]: I would simple function call instead struct, `BindInt` is not reusable anyways, as it's conditions is suitable just for: query params, just for optional ints

**Review by Waldz — CHANGES_REQUESTED** [review]

File: tequilapi/contract/identity.go [hunk 4]
```diff
@@ -129,6 +130,25 @@ type IdentityRegisterRequest struct {
 	ReferralToken *string `json:"token,omitempty"`
 }
 
+// Validate validates fields in request
+func (irr *IdentityRegisterRequest) Validate() *validation.FieldErrorMap {
+	errors := validation.NewErrorMap()
+
+	if irr.Beneficiary != "" && !common.IsHexAddress(irr.Beneficiary) {
```
Comment (Waldz) [inline tequilapi/contract/identity.go:137]: Would be nice to extract this to reusable `BindEthAddress()`

File: tequilapi/contract/identity.go (hunk 2 above)
Comment (vkuznecovas) [reply tequilapi/contract/identity.go:144]: There's no artificial limit imposed.

**Review by vkuznecovas — APPROVED** [review]

**Review by tadaskay — APPROVED** [review]

**Review by soffokl — APPROVED** [review]

**Review by Waldz — APPROVED** [review]

**Comment by chompomonim** [comment]
Why @zolia review is blocking this? We already have a lot of reviews.

**Pull request merged**